### DISCUSSION TITLE
oAuth procedure ends with no result

### DISCUSSION THREAD

**Discussion by IlyaPetrovM** [directus/directus#8823] [Archive]
1. I press login with google
![image](https://user-images.githubusercontent.com/6277090/137462474-8b6a0f57-7edf-4a55-a5f8-f79d3a1e5cc1.png)
2. Go to auth page
3. Enter login and password
4. Browser forwards me to login page

**Comment by joselcvarela** [comment]
Hello @IlyaPetrovM ,
In order to help you, we need more information.
Try open your browser devtools, check Network tab, click on Preserve Log and check if some request fails.
Also, check if redirect_uri on URL matches with what is supposed.

Check this issues and discussions too:
https://github.com/directus/directus/search?q=oauth+google&type=issues
https://github.com/directus/directus/search?q=oauth+google&type=discussions

**Reply by aidenfoxx** [reply]
@IlyaPetrovM Also, what browser are you on? Firefox can have issues with SameSite cookies when using OAuth with Directus.

**Reply by IlyaPetrovM** [reply]
Hi, @joselcvarela!
All requests have status "200".
Did you change `redirect_uri` ? I used `redirect_uri=https://t1553.ru/admin/login` in v9.0.0-rc.96
I use Chrome browser.

**Reply by aidenfoxx** [reply]
The oauth flow is handled by a library called grant, and that's not changed. The redirect_uri appears to be incorrect. It should be something like `/auth/oauth/<provider>/callback`

**Reply by IlyaPetrovM** [reply]
@aidenfoxx 
Well, redirect_uri in google console already was like it has to be:
![image](https://user-images.githubusercontent.com/6277090/137505201-792b0f99-0a17-4c3c-b19b-91ad349860fa.png)

I mean login button. It passes different address:
```html
<a class="sso-link" href="/auth/oauth/google?redirect=https://t1553.ru/admin/login" data-v-09e14c2a="">Log In with google</a>
```

**Reply by aidenfoxx** [reply]
Oh, you're talking about THAT redirect. That's not an OAuth `redirect_uri`. `redirect` is the URL Directus stores internally to return you to the correct url after the OAuth callback. That being said, everything looks correct.

When you redirect back do you see any message in the login URL? Like "SERVICE_UNAVAILABLE" or "UNKNOWN_EXCEPTION"?

**Reply by IlyaPetrovM** [reply]
@aidenfoxx 
after login I see this 
![image](https://user-images.githubusercontent.com/6277090/137509260-0530a118-65e8-4967-88d0-127a5b4bf173.png)

I found request with status 302 at Network tab:
```
https://t1553.ru/auth/oauth/google/callback?code=4/{ 
 secure_code  }&scope=email%20https://www.googleapis.com/auth/userinfo.email%20openid&authuser=0&hd=1553.ru&prompt=none
```
![image](https://user-images.githubusercontent.com/6277090/137509833-2e5d27d3-a46d-4dc9-9b62-f389425e9026.png)

**Reply by aidenfoxx** [reply]
302 just indicates a redirect. Hm... If there was a problem the login page would have an error in the URL.

Are you sure you have a user in Directus with a matching email to the one you're logging in with at google? 🤔

**Reply by IlyaPetrovM** [reply]
I'm sure! And everything worked in rc.96.

**Reply by IlyaPetrovM** [reply]
Thanks to everybody! oAuth now works in `9.0.0-rc.98`!
